### ISSUE TITLE
[BUG] vela up segmentation violation

### ISSUE THREAD

**Issue by hprotzek** [kubevela/kubevela#3805]
Running:
CLI Version: 1.3.3
Core Version: 1.3.3
GitRevision: 45e1de19dc736b10b3eb2d908d809210f470b24f
GolangVersion: go1.18.1

On `vela up -f ./vela.yaml` the command fails with:
```
Parsing vela application file ...
Load Template ...
parsing application componentspanic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x156750a]

goroutine 1 [running]:
github.com/modern-go/reflect2.(*frozenConfig).Type2(0x2f15a80, {0x382cde0, 0x2e25780})
        github.com/modern-go/reflect2@v1.0.1/reflect2.go:157 +0x4a
github.com/modern-go/reflect2.(*UnsafeMapType).Key(0xc000c293c0)
        github.com/modern-go/reflect2@v1.0.1/unsafe_map.go:53 +0x49
github.com/json-iterator/go.decoderOfMap(0xc000f2ee70, {0x38292e8?, 0xc000c293c0})
        github.com/json-iterator/go@v1.1.11/reflect_map.go:14 +0xe5
github.com/json-iterator/go._createDecoderOfType(0xc000f2ee70, {0x38292e8?, 0xc000c293c0})
        github.com/json-iterator/go@v1.1.11/reflect.go:182 +0x2c8
github.com/json-iterator/go.createDecoderOfType(0xc000f2ee70, {0x38292e8, 0xc000c293c0})
        github.com/json-iterator/go@v1.1.11/reflect.go:142 +0xdf
github.com/json-iterator/go.decoderOfType(0xc000f2ee70, {0x38292e8, 0xc000c293c0})
        github.com/json-iterator/go@v1.1.11/reflect.go:124 +0x45
github.com/json-iterator/go.describeStruct(0xc000f2ea50, {0x38294c8?, 0xc0010333e0})
        github.com/json-iterator/go@v1.1.11/reflect_extension.go:380 +0x6d0
github.com/json-iterator/go.decoderOfStruct(0xc000f2ea50, {0x38294c8, 0xc0010333e0})
        github.com/json-iterator/go@v1.1.11/reflect_struct_decoder.go:14 +0xb4
github.com/json-iterator/go._createDecoderOfType(0xc000f2ea50, {0x38294c8?, 0xc0010333e0})
        github.com/json-iterator/go@v1.1.11/reflect.go:176 +0x410
github.com/json-iterator/go.createDecoderOfType(0xc000f2ea50, {0x38294c8, 0xc0010333e0})
        github.com/json-iterator/go@v1.1.11/reflect.go:142 +0xdf
github.com/json-iterator/go.decoderOfType(0xc000f2ea50, {0x38294c8, 0xc0010333e0})
        github.com/json-iterator/go@v1.1.11/reflect.go:124 +0x45
github.com/json-iterator/go.describeStruct(0xc0010b43c8, {0x38294c8?, 0xc001033050})
        github.com/json-iterator/go@v1.1.11/reflect_extension.go:380 +0x6d0
github.com/json-iterator/go.decoderOfStruct(0xc0010b43c8, {0x38294c8, 0xc001033050})
        github.com/json-iterator/go@v1.1.11/reflect_struct_decoder.go:14 +0xb4
github.com/json-iterator/go._createDecoderOfType(0xc0010b43c8, {0x38294c8?, 0xc001033050})
        github.com/json-iterator/go@v1.1.11/reflect.go:176 +0x410
github.com/json-iterator/go.createDecoderOfType(0xc0010b43c8, {0x38294c8, 0xc001033050})
        github.com/json-iterator/go@v1.1.11/reflect.go:142 +0xdf
github.com/json-iterator/go.decoderOfType(0xc0010b43c8, {0x38294c8, 0xc001033050})
        github.com/json-iterator/go@v1.1.11/reflect.go:124 +0x45
github.com/json-iterator/go.(*frozenConfig).DecoderOf(0xc0003a5400, {0x3829388, 0xc0010330b0})
        github.com/json-iterator/go@v1.1.11/reflect.go:114 +0x108
github.com/json-iterator/go.(*Iterator).ReadVal(0xc000a4e2d0, {0x328c5a0, 0xc0012b81e0})
        github.com/json-iterator/go@v1.1.11/reflect.go:72 +0x10b
github.com/json-iterator/go.(*frozenConfig).Unmarshal(0xc0003a5400, {0xc0013ca000?, 0xc000a24385?, 0x7?}, {0x328c5a0, 0xc0012b81e0})
        github.com/json-iterator/go@v1.1.11/config.go:348 +0xa5
k8s.io/apimachinery/pkg/runtime/serializer/json.(*Serializer).Decode(0xc00047cfa0, {0xc0013ca000, 0x2bcc, 0x3000}, 0x0, {0x37fe8b8?, 0xc0012b81e0?})
        k8s.io/apimachinery@v0.22.1/pkg/runtime/serializer/json/json.go:264 +0x84c
k8s.io/apimachinery/pkg/runtime.WithoutVersionDecoder.Decode({{0x37f7c40?, 0xc00047cfa0?}}, {0xc0013ca000?, 0xc0010b48a8?, 0x1cfc980?}, 0xc0010b4888?, {0x37fe8b8?, 0xc0012b81e0?})
        k8s.io/apimachinery@v0.22.1/pkg/runtime/helper.go:252 +0x55
k8s.io/client-go/rest.Result.Into({{0xc0013ca000, 0x2bcc, 0x3000}, {0x0, 0x0, 0x0}, {0xc0009a0980, 0x10}, {0x0, 0x0}, ...}, ...)
        k8s.io/client-go@v0.22.1/rest/request.go:1334 +0xad
sigs.k8s.io/controller-runtime/pkg/client.(*typedClient).Get(0x3278e60?, {0x3816990, 0xc000f2e090}, {{0xc00154e680?, 0x0?}, {0xc00154ecd0?, 0x199?}}, {0x382fd10?, 0xc0012b81e0})
        sigs.k8s.io/controller-runtime@v0.9.5/pkg/client/typed_client.go:143 +0x205
sigs.k8s.io/controller-runtime/pkg/client.(*client).Get(0x3816990?, {0x3816990?, 0xc000f2e090?}, {{0xc00154e680?, 0xc00154ecd0?}, {0xc00154ecd0?, 0x37ee268?}}, {0x382fd10?, 0xc0012b81e0?})
        sigs.k8s.io/controller-runtime@v0.9.5/pkg/client/client.go:253 +0x2d3
github.com/oam-dev/kubevela/pkg/oam/util.GetDefinition({0x3816990, 0xc000f2e090}, {0x6625528, 0xc0002e0e00}, {0x382fd10, 0xc0012b81e0}, {0xc00154ecd0, 0xb})
        github.com/oam-dev/kubevela/pkg/oam/util/helper.go:340 +0x15e
github.com/oam-dev/kubevela/pkg/oam/util.GetCapabilityDefinition({0x3816990, 0xc000f2e090}, {0x6625528, 0xc0002e0e00}, {0x382fd10?, 0xc0012b81e0?}, {0xc00154ecd0, 0xb})
        github.com/oam-dev/kubevela/pkg/oam/util/helper.go:368 +0x2a8
github.com/oam-dev/kubevela/pkg/appfile.LoadTemplate({0x3816990, 0xc000f2e090}, {0x3819bf0, 0xc0010c1f20}, {0x6625528, 0xc0002e0e00}, {0xc00154ecd0, 0xb}, {0x330982f, 0x13})
        github.com/oam-dev/kubevela/pkg/appfile/template.go:77 +0x7d7
github.com/oam-dev/kubevela/pkg/appfile.TemplateLoaderFn.LoadTemplate(...)
        github.com/oam-dev/kubevela/pkg/appfile/parser.go:56
github.com/oam-dev/kubevela/pkg/appfile.(*Parser).makeWorkload(0xc000f2e060, {0x3816990, 0xc000f2e090}, {0xc00154ece0, 0xc}, {0xc00154ecd0, 0xb}, {0x330982f, 0x13}, 0xc000fb1470)
        github.com/oam-dev/kubevela/pkg/appfile/parser.go:435 +0x103
github.com/oam-dev/kubevela/pkg/appfile.(*Parser).parseWorkload(0xc000f2e060, {0x3816990, 0xc000f2e090}, {{0xc00154ece0, 0xc}, {0xc00154ecd0, 0xb}, {0x0, 0x0}, 0xc000fb1470, ...})
        github.com/oam-dev/kubevela/pkg/appfile/parser.go:475 +0x7f
github.com/oam-dev/kubevela/pkg/appfile.(*Parser).GenerateAppFileFromApp(0xc000f2e060, {0x3816990, 0xc000f2e090}, 0xc001286800)
        github.com/oam-dev/kubevela/pkg/appfile/parser.go:117 +0xab0
github.com/oam-dev/kubevela/pkg/appfile.(*Parser).GenerateAppFile(0x3816920?, {0x3816990, 0xc000f2e090}, 0xc001286800)
        github.com/oam-dev/kubevela/pkg/appfile/parser.go:102 +0x2ca
github.com/oam-dev/kubevela/references/appfile.ApplyTerraform(0xc001286800, {0x3824d10?, 0xc0002e0e00}, {{0x37f8b00, 0xc00000e010}, {0x37f85a0, 0xc000692120}, {0x37f8b40, 0xc00000e020}}, {0xc00154e680, ...}, ...)
        github.com/oam-dev/kubevela/references/appfile/addon.go:66 +0x1ee
github.com/oam-dev/kubevela/references/common.(*AppfileOptions).BaseAppFileRun(0xc0010b5cc0, 0xc001b815f0, {0x0, 0xc00046a000, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x0})
        github.com/oam-dev/kubevela/references/common/application.go:379 +0xe5
github.com/oam-dev/kubevela/references/common.(*AppfileOptions).Run(0xc000070000?, {0x7ff7bfeff9a3?, 0x2e25780?}, {0xc00154e680?, 0x66136d0?}, {0x0, 0xc00046a000, {0x0, 0x0}, {0x0, ...}, ...})
        github.com/oam-dev/kubevela/references/common/application.go:373 +0xbd
github.com/oam-dev/kubevela/references/cli.(*UpCommandOptions).deployApplicationFromFile(0xc0002e85f0, {0x37f6020?, 0xc000138220?}, 0xc000efbd20?)
        github.com/oam-dev/kubevela/references/cli/up.go:195 +0x4a9
github.com/oam-dev/kubevela/references/cli.(*UpCommandOptions).Run(0x0?, {0x37f6020?, 0xc000138220?}, 0x21c61c0?)
        github.com/oam-dev/kubevela/references/cli/up.go:82 +0x54
github.com/oam-dev/kubevela/references/cli.NewUpCommand.func2(0xc000e3e000?, {0xc0000c1500?, 0x0?, 0x2?})
        github.com/oam-dev/kubevela/references/cli/up.go:269 +0xb7
github.com/spf13/cobra.(*Command).execute(0xc000e3e000, {0xc0000c14e0, 0x2, 0x2})
        github.com/spf13/cobra@v1.2.1/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0xc000b34c80)
        github.com/spf13/cobra@v1.2.1/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main()
        github.com/oam-dev/kubevela/references/cmd/cli/main.go:42 +0x125
```

Possible thats this dependency thats not updated?
https://github.com/modern-go/reflect2/issues/26

**Comment by hprotzek** [comment]
Reverting vela cli to 1.3.2 fixes the problem.

**Comment by wonderflow** [comment]
@hprotzek Thanks for reporting, we'll fix it ASAP.

**Comment by Somefive** [comment]
@hprotzek Thanks for the reporting. Could you give an example of your ./vela.yaml so that we can diagnose the bug?

**Comment by hprotzek** [comment]
@Somefive 
```
name: snoauth-test
services:
  snoauth-test:
    type: snstateless
    image: stefanprodan/podinfo
    ports:
      - name: http
        containerPort: 9898
        servicePort: 9898
        protocol: http
    sningress:
      routes:
        - route: snoauth-test.apps
          servicePort: 9898
```
```
import "strings"

snstateless: {
	type:        "component"
	description: "Component provides a stateless service to run webapps or worker apps. (v1alpha1)"
	labels: {}
	annotations: {}
	attributes: {
		workload: definition: {
			apiVersion: "apps/v1"
			kind:       "Deployment"
		}
		status: {
			customStatus: #"""
				import "strconv"
				ready: {
					if context.output.status.readyReplicas == _|_ {
						readyReplicas: 0
					}

					if context.output.status.readyReplicas != _|_ {
						readyReplicas: context.output.status.readyReplicas
					}
				}

				message: "Ready:" + strconv.FormatInt(ready.readyReplicas, 10) + "/" + strconv.FormatInt(context.output.spec.replicas, 10)
				"""#
			healthPolicy: #"""
				ready: {
					if context.output.status.updatedReplicas == _|_  {
						updatedReplicas : 0
					}

					if context.output.status.updatedReplicas != _|_  {
						updatedReplicas : context.output.status.updatedReplicas
					}

					if context.output.status.readyReplicas == _|_ {
						readyReplicas: 0
					}

					if context.output.status.readyReplicas != _|_ {
						readyReplicas: context.output.status.readyReplicas
					}

					if context.output.status.replicas == _|_ {
						replicas: 0
					}
					if context.output.status.replicas != _|_ {
						replicas: context.output.status.replicas
					}

					if context.output.status.observedGeneration != _|_ {
						observedGeneration: context.output.status.observedGeneration
					}

					if context.output.status.observedGeneration == _|_ {
						observedGeneration: 0
					}
				}
					isHealth: (context.output.spec.replicas == ready.readyReplicas) && (context.output.spec.replicas == ready.updatedReplicas) && (context.output.spec.replicas == ready.replicas) && (ready.observedGeneration == context.output.metadata.generation || ready.observedGeneration > context.output.metadata.generation)
				"""#
		}
	}
}

template: {
	output: {
		apiVersion: "apps/v1"
		kind:       "Deployment"
		metadata: {
			name: context.name
			labels: {
				app:     context.name
				version: "v1"
			}
		}
		spec: {
			replicas: parameter.replicas
			selector: matchLabels: {
				"app": context.name
			}
			template: {
				metadata: labels: {
					"app":                     context.name
					"version":                 "v1"
					"app.oam.dev/component":   context.name
					"app.oam.dev/revision":    context.revision
					"sidecar.istio.io/inject": "true"
				}
				spec: {
					topologySpreadConstraints: [
						{
							maxSkew:           1
							topologyKey:       "topology.kubernetes.io/zone"
							whenUnsatisfiable: "DoNotSchedule"
							labelSelector: matchLabels: app: context.name
						},
						{
							maxSkew:           1
							topologyKey:       "kubernetes.io/hostname"
							whenUnsatisfiable: "ScheduleAnyway"
							labelSelector: matchLabels: app: context.name
						},
					]
					if parameter.gpuSupport == true {
						nodeSelector: "cloud.google.com/gke-accelerator": parameter.gpuType
					}
					serviceAccountName: "\(context.namespace)"
					containers: [{
						name:            context.name
						image:           parameter.image
						imagePullPolicy: parameter.imagePullPolicy
						if parameter["cmd"] != _|_ {
							command: parameter.cmd
						}
						if parameter["args"] != _|_ {
							args: parameter.args
						}
						if parameter["env"] != _|_ {
							env: parameter.env
						}
						if parameter["ports"] != _|_ {
							ports:
							[ for p in parameter.ports {
								containerPort: p.containerPort
								name:          strings.ToLower("\(p.protocol)-\(p.name)")
								protocol:      strings.ToUpper(strings.Replace(p.protocol, "http", "tcp", 1))
							}]
						}
						if parameter["volumes"] != _|_ {
							volumeMounts: [ for v in parameter.volumes {
								{
									mountPath: v.path
									name:      v.name
								}
							}]
						}
						lifecycle: {
							if parameter["postStartCommand"] != _|_ {
								postStart: exec: command: parameter.postStartCommand
							}
							if parameter["preStopCommand"] != _|_ {
								preStop: exec: command: parameter.preStopCommand
							}
						}
						resources: {
							limits: {
								cpu:    parameter.resources.cpu
								memory: parameter.resources.memory
								if parameter.gpuSupport == true {
									"nvidia.com/gpu": 1
								}
							}
							requests: {
								cpu:    parameter.resources.cpu
								memory: parameter.resources.memory
							}
						}
					}]
					if parameter.gpuSupport == true {
						tolerations: [
							{
								key:      "katee-gpu"
								operator: "Equal"
								value:    "dedicated"
								effect:   "NoSchedule"
							},
							{
								key:      "nvidia.com/gpu"
								operator: "Equal"
								value:    "present"
								effect:   "NoSchedule"
							},
						]
						affinity: {
							nodeAffinity: requiredDuringSchedulingIgnoredDuringExecution: nodeSelectorTerms: [{
								matchExpressions: [
									{
										key:      "katee-gpu"
										operator: "In"
										values: ["dedicated"]
									},
									{
										key:      "cloud.google.com/machine-family"
										operator: "In"
										values: ["n1"]
									},
								]
							}]
						}
					}
					if parameter.gpuSupport == false {
						tolerations: [{
							key:      "katee"
							operator: "Equal"
							value:    "dedicated"
							effect:   "NoSchedule"
						}]
						affinity: nodeAffinity: requiredDuringSchedulingIgnoredDuringExecution: nodeSelectorTerms: [{
							matchExpressions: [{
								key:      "katee"
								operator: "In"
								values: ["dedicated"]
							}]
						}]
					}
				}
			}
		}
	}
	outputs: {
		if parameter.replicas > 1 {
			poddisruptionbudget: {
				apiVersion: "policy/v1"
				kind:       "PodDisruptionBudget"
				metadata: name: context.name
				spec: {
					maxUnavailable: 1
					selector: matchLabels: app: context.name
				}
			}
		}
		if parameter["ports"] != _|_ {
			service: {
				apiVersion: "v1"
				kind:       "Service"
				metadata: {
					name: context.name
					labels: {
						"app":     context.name
						"version": "v1"
					}
				}
				spec: {
					selector: {
						"app.oam.dev/component": context.name
					}
					ports:
					[ for p in parameter.ports {
						port:       p.servicePort
						targetPort: p.containerPort
						name:       strings.ToLower("\(p.protocol)-\(p.name)")
						protocol:   strings.ToUpper(strings.Replace(p.protocol, "http", "tcp", 1))
					}]
				}
			}
		}
	}
	parameter: {
		// +usage=Container image would you like to use for your service with format registry.url/TEAM/name:tag-version
		image: string

		// +usage=When the container image should be pulled from registry, default policy is IfNotPresent. Use Always if you re-push changes with the same tag/versions
		imagePullPolicy: *"IfNotPresent" | "Always"

		// +usage=Number of instances
		replicas: *1 | int

		// +usage=CPU and Memory assigned to each instance
		resources: {
			cpu:    *"200m" | string | int
			memory: *"128Mi" | string | int
		}

		// +usage=Listening ports for incoming traffic
		ports?: [...{
			// +usage=Name of the port
			name: string
			// +usage=Container port number
			containerPort: int
			// +usage=Service port number
			servicePort: int
			// +usage=Protocol
			protocol: *"http" | "tcp" | "udp"
		}]

		// +usage=Startup commands to run in the container, how to run the application.
		cmd?: [...string]

		// +usage=Command arguments
		args?: [...string]

		postStartCommand?: [...string]

		preStopCommand?: [...string]

		env?: [...{
			// +usage=Environment variable name
			name: string
			// +usage=The value of the environment variable
			value?: string
			// +usage=Specifies a source the value of this var should come from
			valueFrom?: {
				// +usage=Selects a key of a secret in the pod's namespace
				secretKeyRef?: {
					// +usage=The name of the secret in the pod's namespace to select from
					name: string
					// +usage=The key of the secret to select from. Must be a valid secret key
					key: string
				}
				// +usage=Selects a key of a config map in the pod's namespace
				configMapKeyRef?: {
					// +usage=The name of the config map in the pod's namespace to select from
					name: string
					// +usage=The key of the config map to select from. Must be a valid secret key
					key: string
				}
			}
		}]

		// +usage=Specify the shared volume path
		volumes?: [...{
			name: string
			path: string
		}]

		// +usage=If enabled the app runs on a node with gpu support. (nvidia-tesla-t4)
		gpuSupport: *false | bool
		gpuType:    "nvidia-tesla-k80" | "nvidia-tesla-p100" | "nvidia-tesla-p4" | "nvidia-tesla-v100" | *"nvidia-tesla-t4" | "nvidia-tesla-a100"
	}
}

```

```
sningress: {
	type:        "trait"
	description: "Create ingress routes. (v1alpha1)"
	labels: {}
	annotations: {}
	attributes: {
		podDisruptive: false
		appliesToWorkloads: ["snstateless"]
		status: {
			customStatus: #"""
				import "strings"
				message: strings.Join([ for r in parameter.routes { "\n\troute: " + r.route + "\n\tservicePort: \(r.servicePort)\n" } ], "")
				"""#
		}
	}
}

template: {
	outputs: {
		for i, r in parameter.routes {
			"virtualservice-\(i)": {
				apiVersion: "networking.istio.io/v1beta1"
				kind:       "VirtualService"
				metadata: name: "\(context.name)-\(i)"
				spec: {
					if r.cname != _|_ {
						hosts: [r.route, r.cname]
					}
					if r.cname == _|_ {
						hosts: [r.route]
					}
					gateways: [
						"mesh",
						"istio-system/private",
						"istio-system/public",
					]
					http: [
						{
							name: "\(context.name)-\(i)-\(context.namespace)-ingress"
							route: [{
								destination: {
									host: "\(context.name).\(context.namespace).svc.cluster.local"
									port: number: r.servicePort
								}
							}]
						},
					]
				}
			}
		}
		for i, r in parameter.routes if r.cname != _|_ {
			"cname-service-\(i)": {
				apiVersion: "v1"
				kind:       "Service"
				metadata: {
					name: "\(context.name)-cname-\(i)"
					annotations: {
						"external-dns.alpha.kubernetes.io/hostname": r.cname
					}
				}
				spec: {
					type:         "ExternalName"
					externalName: r.route
				}
			}
		}
		for i, r in parameter.routes {
			"virtualservice-\(i)": {
				apiVersion: "networking.istio.io/v1beta1"
				kind:       "VirtualService"
				metadata: name: "\(context.name)-\(i)"
				spec: {
					if r.cname != _|_ {
						hosts: [r.route, r.cname]
					}
					if r.cname == _|_ {
						hosts: [r.route]
					}
					gateways: [
						"mesh",
						"istio-system/private",
						"istio-system/public",
					]
					http: [
						{
							name: "\(context.name)-\(i)-\(context.namespace)-ingress"
							route: [{
								destination: {
									host: "\(context.name).\(context.namespace).svc.cluster.local"
									port: number: r.servicePort
								}
							}]
						},
					]
				}
			}
		}
		for i, r in parameter.routes if r.probeEnabled {
			"prometheus-probe-\(i)":{
				apiVersion: "monitoring.coreos.com/v1"
				kind: "Probe"
				metadata: name: "\(context.name)-\(i)"
				spec: {
					jobName: "ingress-probe"
  				interval: "60s"
  				module: "http_2xx"
  				prober: url: "blackbox-exporter.system.svc.cluster.local:9115"
  				targets: {
    				staticConfig: {
      				static: [
      					"https://\(r.route)"
      				]
      			}
      		}
      	}
			}
		}
		for i, r in parameter.routes if r.probeEnabled {
			"prometheus-rule-\(i)": {
				apiVersion: "monitoring.coreos.com/v1"
				kind: "PrometheusRule"
				metadata: {
  				name: "\(context.name)-\(i)"
  				labels: {
    				prometheus: "prometheus-operator-prometheus"
    				role: "alert-rules"
    			}
    		}
				spec: {
					groups: [{
						name: "./katee-\(context.name)-\(i).rules"
						rules: [{
							alert: "\(context.name)-\(i)"
							expr: "probe_success{job=\"ingress-probe\", instance=\"https://\(r.route)\"} * on (namespace) group_left(annotation_alert_email, annotation_alert_slack) kube_namespace_annotations == 0"
							for: "5m"
							labels: {
								severity: "critical"
								namespace: context.namespace
							}
							annotations: {
								summary: "Ingress is down."
								description: "\(context.name) ingress {{`{{ $labels.instance }}`}} from is down."
							}
						}]
					}]
				}
			}
		}
	}
	parameter: {
		// +usage=Route to access HTTP service
		routes: [...{
			route:       string
			cname?:      string
			servicePort: int
			// +usage=Creates a probe to check availability of the route endpoint. Alerts are send to the configured team slack or email channels.
			probeEnabled: true | *false
		}]
	}
}

```

**Comment by Somefive** [comment]
@hprotzek Thanks for the example. I will look into it soon.

**Comment by Somefive** [comment]
@hprotzek Sorry I wasn't able to reproduce the reported problems correctly. The vela 1.3.3 with the described environment actually works properly. It works as follows

```bash
➜  v1.3.3 ./vela up -f ../../3805/appfile.yaml
Parsing vela application file ...
Load Template ...
parsing application components
Applying application ...
Checking if app has been deployed...
App has not been deployed, creating a new deployment...
✅ App has been deployed 🚀🚀🚀
    Port forward: vela port-forward snoauth-test
             SSH: vela exec snoauth-test
         Logging: vela logs snoauth-test
      App status: vela status snoauth-test
        Endpoint: vela status snoauth-test --endpoint
```

There are several things that we could explore more.
1. Although I tried the vela-1.3.3 CLI for testing, it is a bit weird that the GoLang version in your vela-v1.3.3 is 1.18.1, since KubeVela now mainly uses 1.17 for build & testing. Which particular binary of vela CLI did you use? Did you build vela CLI on your own?
2. You said 1.3.2 works properly and I checked the changes from 1.3.2 to 1.3.3, seems no changes related to the code here.
3. 1.3.3 does use old reflect2 library which could lead to some potential problems, but I'm not very sure if it is related to this issue. The master branch of KubeVela upgrade the reflect2 dependencies recently. You can try to use the master branch or wait for 1.3.4 release (it should be release in one or two days).
4. We recommend to use the Application other than AppFile after KubeVela v1.2, which provides more powerful functionalities. The AppFile part now is mainly maintained for compatibility. Though, this recommendation I think is not related to this issue.

**Comment by hprotzek** [comment]
Not sure how I installed the not functional version, but installing it now with brew, the 1.3.3 works and also show the correct git revision.
```
GitRevision: 45e1de1
```
in comparison to my faulty one
```
GitRevision: git-45e1de1
```

I can't remember that I build it locally, but I tried it now and the build failed because of lint errors, so I don't think it was a locally build version. Anyway, the version installed with brew works now.
